include/irr/asset/ICPUBuffer.h

```diff
@@ -97,21 +97,26 @@ class ICPUBuffer : public core::IBuffer, public asset::IAsset
 };
 
 template<typename Allocator = _IRR_DEFAULT_ALLOCATOR_METATYPE<uint8_t>>
-class IMemoryAdoptingCPUBuffer : public ICPUBuffer
+class CCustomAllocatorCPUBuffer : public ICPUBuffer
 {
+    static_assert(sizeof(Allocator::value_type)==1u, "Allocator::value_type must be of size 1");
 protected:
     Allocator m_allocator;
 
-    virtual ~IMemoryAdoptingCPUBuffer()
+    virtual ~CCustomAllocatorCPUBuffer()
     {
         if (ICPUBuffer::data)
-            m_allocator.deallocate(reinterpret_cast<typename Allocator::pointer>(ICPUBuffer::data), ICPUBuffer::size/sizeof(typename Allocator::value_type));
-        ICPUBuffer::data = nullptr; // so that ICPUBuffer will not try deallocating
+            m_allocator.deallocate(reinterpret_cast<typename Allocator::pointer>(ICPUBuffer::data), ICPUBuffer::size);
+        ICPUBuffer::data = nullptr;
     }
 
 public:
-    IMemoryAdoptingCPUBuffer(size_t sizeInBytes, void* dat, Allocator&& allctr = Allocator()) : ICPUBuffer(sizeInBytes, dat), m_allocator(std::move(allctr))
-    {}
+    CCustomAllocatorCPUBuffer(size_t sizeInBytes, void* dat, Allocator&& alctr = Allocator()) : CCustomAllocatorCPUBuffer(sizeInBytes, alctr.allocate(allocSz), core::adopt_memory, std::move(alctr))

```

Comment:
why is this one implicitly adopting memory?

Reply:
just using the adopting constructor but as you can see it allocates its own mem. However it's missing memcpy